File: packages/go/domain/event.go
package domain

import (
	"time"
)

type EventType string

type EventEnvelope struct {
	EventID        string      `json:"event_id"`
	EventType      EventType   `json:"event_type"`
	EventVersion   string      `json:"event_version"`
	OccurredAt     time.Time   `json:"occurred_at"`
	Producer       string      `json:"producer"`
	TenantID       *string     `json:"tenant_id,omitempty"`
	UserID         *string     `json:"user_id,omitempty"`
	BotID          *string     `json:"bot_id,omitempty"`
	CorrelationID  string      `json:"correlation_id"`
	CausationID    *string     `json:"causation_id,omitempty"`
	IdempotencyKey string      `json:"idempotency_key"`
	Payload        interface{} `json:"payload"`
}

const (
	// Bot events
	EventTypeBotCreated   EventType = "bot.created"
	EventTypeBotActivated EventType = "bot.activated"
	EventTypeBotPaused    EventType = "bot.paused"
	EventTypeBotStopped   EventType = "bot.stopped"
	EventTypeBotError     EventType = "bot.error"

	// Subscription events
	EventTypeSubscriptionVerified EventType = "subscription.verified"

	// Trading events
	EventTypeSignalGenerated          EventType = "signal.generated"
	EventTypeRiskCheckRequested       EventType = "risk.check.requested"
	EventTypeRiskCheckPassed          EventType = "risk.check.passed"
	EventTypeRiskCheckFailed          EventType = "risk.check.failed"
	EventTypeOrderIntentCreated       EventType = "order.intent.created"
	EventTypeOrderSubmitted           EventType = "order.submitted"
	EventTypeOrderAccepted            EventType = "order.accepted"
	EventTypeOrderRejected            EventType = "order.rejected"
	EventTypeExecutionFilled          EventType = "execution.filled"
	EventTypeExecutionPartiallyFilled EventType = "execution.partially_filled"
	EventTypeExecutionFailed          EventType = "execution.failed"

	// Position events
	EventTypePositionUpdated EventType = "position.updated"
	EventTypePnLUpdated      EventType = "pnl.updated"

	// System events
	EventTypeEmergencyStopActivated EventType = "emergency_stop.activated"
)

File: services/executor-rust/src/types/event.rs
use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct EventEnvelope<T> {
    pub event_id: String,
    pub event_type: String,
    pub event_version: String,
    pub occurred_at: DateTime<Utc>,
    pub producer: String,
    pub tenant_id: Option<String>,
    pub user_id: Option<String>,
    pub bot_id: Option<String>,
    pub correlation_id: String,
    pub causation_id: Option<String>,
    pub idempotency_key: String,
    pub payload: T,
}

pub const EVENT_TYPE_BOT_CREATED: &str = "bot.created";
pub const EVENT_TYPE_BOT_ACTIVATED: &str = "bot.activated";
pub const EVENT_TYPE_BOT_PAUSED: &str = "bot.paused";
pub const EVENT_TYPE_BOT_STOPPED: &str = "bot.stopped";
pub const EVENT_TYPE_BOT_ERROR: &str = "bot.error";

pub const EVENT_TYPE_SUBSCRIPTION_VERIFIED: &str = "subscription.verified";

pub const EVENT_TYPE_SIGNAL_GENERATED: &str = "signal.generated";
pub const EVENT_TYPE_RISK_CHECK_REQUESTED: &str = "risk.check.requested";
pub const EVENT_TYPE_RISK_CHECK_PASSED: &str = "risk.check.passed";
pub const EVENT_TYPE_RISK_CHECK_FAILED: &str = "risk.check.failed";
pub const EVENT_TYPE_ORDER_INTENT_CREATED: &str = "order.intent.created";
pub const EVENT_TYPE_ORDER_SUBMITTED: &str = "order.submitted";
pub const EVENT_TYPE_ORDER_ACCEPTED: &str = "order.accepted";
pub const EVENT_TYPE_ORDER_REJECTED: &str = "order.rejected";
pub const EVENT_TYPE_EXECUTION_FILLED: &str = "execution.filled";
pub const EVENT_TYPE_EXECUTION_PARTIALLY_FILLED: &str = "execution.partially_filled";
pub const EVENT_TYPE_EXECUTION_FAILED: &str = "execution.failed";

pub const EVENT_TYPE_POSITION_UPDATED: &str = "position.updated";
pub const EVENT_TYPE_PNL_UPDATED: &str = "pnl.updated";

pub const EVENT_TYPE_EMERGENCY_STOP_ACTIVATED: &str = "emergency_stop.activated";
